src/react-components/presence-log.js

```diff
@@ -5,6 +5,107 @@ import classNames from "classnames";
 import Linkify from "react-linkify";
 import { toArray as toEmojis } from "react-emoji-render";
 import { FormattedMessage } from "react-intl";
+import serializeElement from "../utils/serialize-element";
+
+const messageCanvas = document.createElement("canvas");
+const presenceLogSpawnedStyle = `background-color: white; color: black; padding: 8px 16px; border-radius: 16px;`;
+const presenceLogPureEmojiStyle = `background-color: transparent; text-align: center;`;
+const emojiRegex = /(?:[\u2700-\u27bf]|(?:\ud83c[\udde6-\uddff]){2}|[\ud800-\udbff][\udc00-\udfff]|[\u0023-\u0039]\ufe0f?\u20e3|\u3299|\u3297|\u303d|\u3030|\u24c2|\ud83c[\udd70-\udd71]|\ud83c[\udd7e-\udd7f]|\ud83c\udd8e|\ud83c[\udd91-\udd9a]|\ud83c[\udde6-\uddff]|[\ud83c[\ude01-\ude02]|\ud83c\ude1a|\ud83c\ude2f|[\ud83c[\ude32-\ude3a]|[\ud83c[\ude50-\ude51]|\u203c|\u2049|[\u25aa-\u25ab]|\u25b6|\u25c0|[\u25fb-\u25fe]|\u00a9|\u00ae|\u2122|\u2139|\ud83c\udc04|[\u2600-\u26FF]|\u2b05|\u2b06|\u2b07|\u2b1b|\u2b1c|\u2b50|\u2b55|\u231a|\u231b|\u2328|\u23cf|[\u23e9-\u23f3]|[\u23f8-\u23fa]|\ud83c\udccf|\u2934|\u2935|[\u2190-\u21ff])/;
+const urlRegex = /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)$/;
+
+function ChatMessage(props) {
+  const ref = React.createRef();
+
+  const handleSpawn = () => {
+    if (props.body.match(urlRegex)) {
+      document.querySelector("a-scene").emit("add_media", props.body);
+      return;
+    }
+
+    const el = ref.current;
```

Comment:
Rather than try and render the current element and doing all this CSS trickery I would maybe just create a new element and then `ReactDOM.render`ing to that. Like 
```
const tmpEl = document.createElement("div");
ReactDOM.render(<ChatMessage {...props} renderForSpawn />, tmpEl, () => { // do SVG stuff with tmpEl })
```

Reply:
going to try this today, if it works I'll add "ctrl-shift enter will do the same thing as enter + hit spawn button, but not echo to chat"